packages/devextreme/testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/mentionModule.tests.js

```diff
@@ -161,32 +163,53 @@ QUnit.module('Mention format', () => {
         };
 
         const element = MentionFormat.create(data);
-        assert.strictEqual(element.innerText, '#John Smith', 'correct inner text');
+
+        new MentionFormat({}, element);
+        assert.strictEqual(element.children[0].innerText, '#John Smith', 'correct inner text');
     });
 
     test('Change default content renderer', function(assert) {
-        const data = {
+        const expectedData = {
             value: 'John Smith',
             marker: '@',
-            id: 'JohnSm',
+            id: 'JohnSm'
+        };
+        const data = {
+            ...expectedData,
             keyInTemplateStorage: 'my_key_in_storage'
         };
 
         MentionFormat.addTemplate({ marker: '@', editorKey: 'my_key_in_storage' }, {
             render: ({ container, model: mentionData }) => {
                 container.innerText = 'test';
-                assert.deepEqual(mentionData, data);
+                assert.deepEqual(mentionData, expectedData);
             }
         });
 
         let element = MentionFormat.create(data);
 
-        assert.strictEqual(element.innerText, 'test');
+        new MentionFormat({}, element);
+        assert.strictEqual(element.children[0].innerText, 'test');
 
         MentionFormat.removeTemplate({ marker: '@', editorKey: 'my_key_in_storage' });
         element = MentionFormat.create(data);
+        new MentionFormat({}, element);
+        assert.strictEqual(element.children[0].innerText, '@John Smith');
+    });
+
+    test('async template renders correctly in React strict mode (T1187274)', function(assert) {
+        let spy;
+        try {
+            const node = MentionFormat.create({ marker: '@', value: 'val', id: 'id' });
+            spy = sinon.spy(MentionFormat.prototype, 'renderContent');
```

Comment:
We try to not use a private api in tests.
I believe there should be a way to test it better.
Give me time to check

Reply:
I did some changes to avoid it.

Reply:
I suggest u to write a more "live" test using this HtmlEditor's configuration:
```javascript
    templatesRenderAsynchronously: true,
    integrationOptions: {
        templates: {
            mention: {
                render: function({ model, container, onRendered }) {
                    setTimeout(() => {
                        const $span = $('<span>');
                        $span.attr({ 'data-id': "custom" });
                        $span.text(`${model.marker}${model.value}`);
                        container.append($span);
                    }, 1000);
                }
            }
        }
    },
```
Your actual code gives this markup:
```html
<span class="dx-mention" spellcheck="false" data-marker="@" data-mention-value="Kevin Carter" data-id="undefined">
    &#xFEFF;
    <span contenteditable="false">
        <span data-attribute="hello">@John Heart</span>
    </span>
    &#xFEFF;
</span>
```
while a code without your fix gives
```html
<span class="dx-mention" spellcheck="false" data-marker="@" data-mention-value="Kevin Carter" data-id="undefined">
    &#xFEFF;
    <span contenteditable="false"></span>
    &#xFEFF;
    <span data-attribute="hello">@Kevin Carter</span>
</span>
```